### PR TITLE
FTP-1377-typography-tests

### PR DESCRIPTION
**Pull request by HamidAli111**
updated

### CODE REVIEW

File: components/heading/Heading.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  it('renders the component', () => {
+    render(<Heading title="Hello" variant="medium-bold" />);
+    const text = screen.getByRole('heading', { level: 1 });
+    expect(text).toHaveClass('medium-bold');
+  });
+  it('renders the component', () => {
+    render(<Heading title="Hello" variant="large-bold" />);
+    const text = screen.getByRole('heading', { level: 1 });
+    expect(text).toHaveClass('large-bold');
+  });
```
Comment (skippednote) [inline components/heading/Heading.spec.tsx:34]: Please make the `it` string verbose

**Review by skippednote — CHANGES_REQUESTED** [review]

File: components/heading/Heading.spec.tsx (hunk 1 above)
Comment (HamidAli111) [reply components/heading/Heading.spec.tsx]: @skippednote Made the required changes and also changed .spec to .test

File: components/text/Text.tsx [hunk 2]
```diff
@@ -4,30 +4,30 @@ import textType from '../../@types/text.types';
...
   if (variant == 'large-bold') {
-    classes = 'font-bold text-xl leading-lg';
+    classes = 'font-bold text-xl leading-lg large-bold';
   }
   if (variant == 'xlarge-bold') {
-    classes = 'font-bold text-2xl leading-xl';
+    classes = 'font-bold text-2xl leading-xl xlarge-bold';
   }
-  return <p className={classes + ' ' + extraClasses}>{title}</p>;
+  return <h2 className={classes + ' ' + extraClasses}>{title}</h2>;
```
Comment (skippednote) [inline components/text/Text.tsx:30]: Let's make this a `p`

File: components/text/Text.tsx [hunk 3]
```diff
@@ -4,30 +4,30 @@ import textType from '../../@types/text.types';
...
-    classes = 'font-bold text-lg leading-sm';
+    classes = 'font-bold text-lg leading-sm medium-bold';
   }
   if (variant == 'large-bold') {
-    classes = 'font-bold text-xl leading-lg';
+    classes = 'font-bold text-xl leading-lg large-bold';
   }
   if (variant == 'xlarge-bold') {
-    classes = 'font-bold text-2xl leading-xl';
+    classes = 'font-bold text-2xl leading-xl xlarge-bold';
```
Comment (skippednote) [inline components/text/Text.tsx:28]: Let's remove the additional classes

**Review by skippednote — CHANGES_REQUESTED** [review]

File: components/heading/Heading.spec.tsx (hunk 1 above)
Comment (HamidAli111) [reply components/heading/Heading.spec.tsx]: @skippednote Made the required changes

File: components/text/Text.tsx (hunk 2 above)
Comment (HamidAli111) [reply components/text/Text.tsx]: @skippednote Made the required changes

File: components/text/Text.tsx (hunk 3 above)
Comment (HamidAli111) [reply components/text/Text.tsx:28]: @skippednote Made the required changes

Comment (HamidAli111) [reply components/text/Text.tsx:28]: Hey @skippednote Not able to see the feedback you gave over the DSM

File: components/heading/Heading.tsx [hunk 4]
```diff
@@ -1,28 +1,27 @@
...
   if (variant == 'medium-bold') {
-    classes = 'font-bold text-4xl leading-3xl'; //20-30
+    classes = 'font-bold text-4xl leading-3xl';
   }
   if (variant == 'large-bold') {
-    classes = 'font-bold text-5xl leading-4xl'; //30-44
+    classes = 'font-bold text-5xl leading-4xl';
   }
-  return <p className={classes + ' ' + extraClasses}>{title}</p>;
+  return <h1 className={classes + ' ' + extraClasses}>{title}</h1>;
```
Comment (skippednote) [inline components/heading/Heading.tsx:25]:
```suggestion
  const Tag = `h${size}` as keyof JSX.IntrinsicElements;
  return <Tag className={classes + ' ' + extraClasses}>{title}</Tag>;
```

**Review by skippednote — CHANGES_REQUESTED** [review]

File: components/heading/Heading.tsx (hunk 4 above)
Comment (HamidAli111) [reply components/heading/Heading.tsx:25]: @skippednote While making the changes I find out that sizes we are already changing text sizes based on the variant, do we still need to implement this way?

Comment (HamidAli111) [reply components/heading/Heading.tsx]: @skippednote Made the required changes

**Pull request merged**